script/watch.js

```diff
@@ -0,0 +1,129 @@
+// babel-register only looks for environment variables
+process.env.BABEL_ENV = process.env.BABEL_ENV || 'test';
+// use babel-register to compile files on the fly
+require('babel-register');
+
+// require mocha setup files
+require('../test/util/mocha-setup.js');
+require('../test/util/enzyme-setup.js');
+
+const { Writable } = require('stream');
+const chokidar = require('chokidar');
+const uncache = require('recursive-uncache');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const glob = util.promisify(require('glob'));
+
+const Mocha = require('mocha');
+
+const chalk = require('chalk');
+const path = require('path');
+const _ = require('lodash');
+
+// no arguments, run unit tests
+// any arguments provited then tests run must be explicitly provided
+let options;
+if (process.argv.length === 2) {
+  options= {
+    eslint: false,
+    sasslint: false,
+    showErrors: false,
+    unit: true
+  };
+} else {
+  options = {
+    eslint: process.argv.includes('eslint'),
+    sasslint: process.argv.includes('sasslint'),
+    showErrors: process.argv.includes('showErrors'),
+    unit: process.argv.includes('unit')
+  };
+}
+
+const unwatchedModules = Object.keys(require.cache).map(file => file);
+let watcher;
+
+async function runUnitTests(tests) {
+  const unitTests = tests || await glob('test/**/*.unit.spec.js?(x)')
+  console.log(chalk.cyan(`Running ${unitTests.length} unit test files`));
+
+  await runMochaTests(unitTests);
+
+  console.log(chalk.red('Watching for changes'));
+
+  if (!watcher) {
+    const watchedFiles = Object.keys(require.cache).filter(file => !unwatchedModules.includes(file) && !file.includes('node_modules'));
+    const unitTestsForSrc = getUnitTestsForSrc(watchedFiles);
+    watcher = chokidar.watch(watchedFiles).
+      on('change', file => {
+        if (file.includes('.unit.spec')) {
+          uncache(file);
+          runUnitTests([file]);
+        }
+
+        if (file.includes('src')) {
```

Comment:
If a unit test is updated, just run that unit test. if a src file is updated, run all the unit tests associated with it.

This will probably go crazy if many files are saved at once but I figure if that's a common scenario we can update this to accommodate.